lib/v2/screens/profile_screens/security/interactor/mappers/guardians_state_mapper.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/v2/datasource/remote/model/firebase_models/guardian_model.dart';
+import 'package:seeds/v2/datasource/remote/model/firebase_models/guardian_status.dart';
+import 'package:seeds/v2/datasource/remote/model/firebase_models/guardian_type.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/profile_screens/security/interactor/viewmodels/security_state.dart';
+import 'package:flutter/cupertino.dart';
+
+class GuardianStateMapper extends StateMapper {
+  SecurityState mapResultToState(bool isGuardianActive, Iterable<GuardianModel> data, SecurityState currentState) {
+    var myGuardians = data.where((element) => element.type == GuardianType.myGuardian);
+    var alreadyGuardians = myGuardians.where((element) => element.status == GuardianStatus.alreadyGuardian);
+
+    Widget guardianStateText;
+
+    if (alreadyGuardians.length < 3) {
+      guardianStateText = const Text(
+        "Inactive",
+        style: TextStyle(color: AppColors.red),
+      );
+    } else {
+      if (isGuardianActive) {
+        guardianStateText = const Text("Active",
+            style: TextStyle(
+              color: AppColors.green1,
+            ));
+      } else {
+        guardianStateText = const Text(
+          "Ready To Activate",
+          style: TextStyle(color: AppColors.orange),
+        );
+      }
+    }
```

Comment:
I would put this logic in the SecurityCard to not return the widget.

I don't like the idea of ​​having widgets in the state at all. 

Maybe you could make a card just for guardians with the logic for the color of the text if the SecurityCard cannot be extended further.

Reply:
agree, no widgets in the state.

Reply:
Thank you both for your review of this pr. 
I added a new component called guardian_security_card. 


https://user-images.githubusercontent.com/51983923/124632392-e91e7b80-de49-11eb-95f9-6b266f53d525.mov